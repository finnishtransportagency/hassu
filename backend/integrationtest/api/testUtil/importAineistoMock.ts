import { SQSEvent, SQSRecord } from "aws-lambda/trigger/sqs";
import * as sinon from "sinon";
import { aineistoImporterClient } from "../../../src/aineisto/aineistoImporterClient";
import { handleEvent } from "../../../src/aineisto/aineistoImporterLambda";
import { Callback, Context } from "aws-lambda";

export class ImportAineistoMock {
  fakeAineistoImportQueue: SQSEvent[] = [];
  importAineistoStub!: sinon.SinonStub;

  initStub(): void {
    this.importAineistoStub = sinon.stub(aineistoImporterClient, "importAineisto");
    this.importAineistoStub.callsFake(async (event) => {
      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
      this.fakeAineistoImportQueue.push({ Records: [{ body: JSON.stringify(event) } as SQSRecord] });
    });
  }

  async processQueue(): Promise<void> {
    for (const event of this.fakeAineistoImportQueue) {
      await handleEvent(event, undefined as unknown as Context, undefined as unknown as Callback);
    }
    this.fakeAineistoImportQueue.splice(0, this.fakeAineistoImportQueue.length); // Clear the queue
  }
}
