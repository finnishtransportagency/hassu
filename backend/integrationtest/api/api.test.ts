import { describe, it } from "mocha";
import { replaceAWSDynamoDBWithLocalstack, setupLocalDatabase } from "../util/databaseUtil";
import { Status } from "../../../common/graphql/apiModel";
import * as sinon from "sinon";
import { personSearchUpdaterClient } from "../../src/personSearch/personSearchUpdaterClient";
import * as personSearchUpdaterHandler from "../../src/personSearch/lambda/personSearchUpdaterHandler";
import { openSearchClientYllapito } from "../../src/projektiSearch/openSearchClient";
import { UserFixture } from "../../test/fixture/userFixture";
import { userService } from "../../src/user";
import { aineistoImporterClient } from "../../src/aineisto/aineistoImporterClient";
import { SQSEvent, SQSRecord } from "aws-lambda/trigger/sqs";
import AWSMock from "aws-sdk-mock";
import AWS from "aws-sdk";
import { getCloudFront, produce } from "../../src/aws/client";
import { cleanProjektiS3Files } from "../util/s3Util";
import { emailClient } from "../../src/email/email";
import {
  archiveProjekti,
  insertAndManageFeedback,
  julkaiseSuunnitteluvaihe,
  julkaiseVuorovaikutus,
  loadProjektiFromDatabase,
  processQueue,
  readProjektiFromVelho,
  sendEmailDigests,
  testAloituskuulutusApproval,
  testAloitusKuulutusEsikatselu,
  testImportAineistot,
  testListDocumentsToImport,
  testNullifyProjektiField,
  testProjektiHenkilot,
  testProjektinTiedot,
  testPublicAccessToProjekti,
  testSuunnitteluvaihePerustiedot,
  testSuunnitteluvaiheVuorovaikutus,
  testUpdatePublishDateAndDeleteAineisto,
  verifyCloudfrontWasInvalidated,
  verifyEmailsSent,
} from "./testUtil/tests";
import { takeS3Snapshot } from "./testUtil/util";
const sandbox = sinon.createSandbox();

describe("Api", () => {
  let readUsersFromSearchUpdaterLambda: sinon.SinonStub;
  let importAineistoStub: sinon.SinonStub;
  let userFixture: UserFixture;
  let oid: string = undefined;
  let awsCloudfrontInvalidationStub: sinon.SinonStub;
  let emailClientStub: sinon.SinonStub;

  after(() => {
    userFixture.logout();
    sandbox.restore();
    sinon.restore();
    AWSMock.restore();
  });

  before("Initialize test database!", async () => {
    await setupLocalDatabase();
  });

  const fakeAineistoImportQueue: SQSEvent[] = [];

  before(async () => {
    userFixture = new UserFixture(userService);
    readUsersFromSearchUpdaterLambda = sandbox.stub(personSearchUpdaterClient, "readUsersFromSearchUpdaterLambda");
    readUsersFromSearchUpdaterLambda.callsFake(async () => {
      return await personSearchUpdaterHandler.handleEvent();
    });

    sandbox.stub(openSearchClientYllapito, "query").resolves({ status: 200 });
    sandbox.stub(openSearchClientYllapito, "deleteProjekti");
    sandbox.stub(openSearchClientYllapito, "putProjekti");

    importAineistoStub = sandbox.stub(aineistoImporterClient, "importAineisto");
    importAineistoStub.callsFake(async (event) => {
      fakeAineistoImportQueue.push({ Records: [{ body: JSON.stringify(event) } as SQSRecord] });
    });

    awsCloudfrontInvalidationStub = sandbox.stub();
    awsCloudfrontInvalidationStub.resolves({});
    AWSMock.setSDKInstance(AWS);
    produce<AWS.CloudFront>("cloudfront", () => undefined, true);
    AWSMock.mock("CloudFront", "createInvalidation", awsCloudfrontInvalidationStub);
    getCloudFront();

    emailClientStub = sinon.stub(emailClient, "sendEmail");
  });

  it("should search, load and save a project", async function () {
    if (process.env.SKIP_VELHO_TESTS == "true") {
      this.skip();
    }
    userFixture.loginAs(UserFixture.mattiMeikalainen);

    const projekti = await readProjektiFromVelho();
    oid = projekti.oid;
    await cleanProjektiS3Files(oid);
    const projektiPaallikko = await testProjektiHenkilot(projekti, oid);
    await testProjektinTiedot(oid);
    await testAloitusKuulutusEsikatselu(oid);
    await testNullifyProjektiField(oid);
    await testAloituskuulutusApproval(oid, projektiPaallikko, userFixture);

    await testSuunnitteluvaihePerustiedot(oid);
    await testSuunnitteluvaiheVuorovaikutus(oid, projektiPaallikko);
    const velhoAineistoKategorias = await testListDocumentsToImport(oid);
    await testImportAineistot(oid, velhoAineistoKategorias);
    await processQueue(fakeAineistoImportQueue, userFixture, oid);
    await testPublicAccessToProjekti(oid, Status.ALOITUSKUULUTUS, userFixture, " ennen suunnitteluvaihetta");

    userFixture.loginAs(UserFixture.mattiMeikalainen);
    await julkaiseSuunnitteluvaihe(oid);
    await loadProjektiFromDatabase(oid, Status.SUUNNITTELU);
    await insertAndManageFeedback(oid);

    await julkaiseVuorovaikutus(oid, userFixture);
    await takeS3Snapshot(oid, "just after vuorovaikutus published");
    verifyCloudfrontWasInvalidated(awsCloudfrontInvalidationStub);

    await testUpdatePublishDateAndDeleteAineisto(oid, userFixture);
    await takeS3Snapshot(oid, "vuorovaikutus publish date changed and last aineisto deleted");
    verifyCloudfrontWasInvalidated(awsCloudfrontInvalidationStub);

    await sendEmailDigests();
    verifyEmailsSent(emailClientStub);
  });

  it.skip("should archive projekti", async function () {
    replaceAWSDynamoDBWithLocalstack();
    await archiveProjekti(oid);
  });
});
