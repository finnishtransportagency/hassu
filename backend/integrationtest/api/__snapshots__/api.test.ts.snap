// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api should search, load and save a project 1`] = `
Object {
  "description": "testSuunnitteluvaihePerustiedot",
  "obj": Object {
    "__typename": "SuunnitteluVaihe",
    "arvioSeuraavanVaiheenAlkamisesta": "huomenna",
    "hankkeenKuvaus": Object {
      "SAAME": "Saameksi suunnitteluvaihe",
      "SUOMI": "Lorem Ipsum suunnitteluvaihe",
      "__typename": "HankkeenKuvaukset",
    },
    "julkinen": undefined,
    "palautteet": undefined,
    "suunnittelunEteneminenJaKesto": "suunnitelma etenee aikataulussa ja valmistuu vuoden 2022 aikana",
    "vuorovaikutukset": Array [],
  },
}
`;

exports[`Api should search, load and save a project 2`] = `
Object {
  "description": "testSuunnitteluvaiheVuorovaikutus",
  "obj": Object {
    "__typename": "SuunnitteluVaihe",
    "arvioSeuraavanVaiheenAlkamisesta": "huomenna",
    "hankkeenKuvaus": Object {
      "SAAME": "Saameksi suunnitteluvaihe",
      "SUOMI": "Lorem Ipsum suunnitteluvaihe",
      "__typename": "HankkeenKuvaukset",
    },
    "julkinen": undefined,
    "palautteet": undefined,
    "suunnittelunEteneminenJaKesto": "suunnitelma etenee aikataulussa ja valmistuu vuoden 2022 aikana",
    "vuorovaikutukset": Array [
      Object {
        "__typename": "Vuorovaikutus",
        "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
        "aineistot": undefined,
        "esitettavatYhteystiedot": Array [
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "Marko",
            "organisaatio": "Kajaani",
            "puhelinnumero": "0293121213",
            "sahkoposti": "markku.koi@koi.com",
            "sukunimi": "Koi",
          },
        ],
        "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
        "videot": Array [
          Object {
            "__typename": "Linkki",
            "nimi": "Esittely 1",
            "url": "https://video",
          },
        ],
        "vuorovaikutusJulkaisuPaiva": "2022-03-23",
        "vuorovaikutusNumero": 1,
        "vuorovaikutusTilaisuudet": Array [
          Object {
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "15:00",
            "kaytettavaPalvelu": "TEAMS",
            "linkki": "https://linkki_tilaisuuteen",
            "nimi": "Lorem ipsum 1",
            "paattymisAika": "16:00",
            "paivamaara": "2022-03-04",
            "tyyppi": "VERKOSSA",
          },
          Object {
            "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "10:00",
            "nimi": "Lorem ipsum two 1",
            "osoite": "Katu 123",
            "paattymisAika": "11:00",
            "paikka": "Kunnantalo",
            "paivamaara": "2022-04-05",
            "postinumero": "00100",
            "postitoimipaikka": "Helsinki",
            "tyyppi": "PAIKALLA",
          },
        ],
        "vuorovaikutusYhteysHenkilot": Array [
          "A000112",
        ],
      },
      Object {
        "__typename": "Vuorovaikutus",
        "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
        "aineistot": undefined,
        "esitettavatYhteystiedot": Array [
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "Marko",
            "organisaatio": "Kajaani",
            "puhelinnumero": "0293121213",
            "sahkoposti": "markku.koi@koi.com",
            "sukunimi": "Koi",
          },
        ],
        "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
        "videot": Array [
          Object {
            "__typename": "Linkki",
            "nimi": "Esittely 2",
            "url": "https://video",
          },
        ],
        "vuorovaikutusJulkaisuPaiva": "2022-03-23",
        "vuorovaikutusNumero": 2,
        "vuorovaikutusTilaisuudet": Array [
          Object {
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "15:00",
            "kaytettavaPalvelu": "TEAMS",
            "linkki": "https://linkki_tilaisuuteen",
            "nimi": "Lorem ipsum 2",
            "paattymisAika": "16:00",
            "paivamaara": "2022-03-04",
            "tyyppi": "VERKOSSA",
          },
          Object {
            "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "10:00",
            "nimi": "Lorem ipsum two 2",
            "osoite": "Katu 123",
            "paattymisAika": "11:00",
            "paikka": "Kunnantalo",
            "paivamaara": "2022-04-05",
            "postinumero": "00100",
            "postitoimipaikka": "Helsinki",
            "tyyppi": "PAIKALLA",
          },
        ],
        "vuorovaikutusYhteysHenkilot": Array [
          "A000112",
        ],
      },
    ],
  },
}
`;

exports[`Api should search, load and save a project 3`] = `
Object {
  "description": "modified testSuunnitteluvaiheVuorovaikutus",
  "obj": "
+         \\"FOO123\\",
",
}
`;

exports[`Api should search, load and save a project 4`] = `
Object {
  "description": "saveAndVerifyAineistoSave",
  "obj": Object {
    "__typename": "Vuorovaikutus",
    "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
    "aineistot": Array [
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.2303050596.2160380418",
        "jarjestys": 1,
        "kategoria": "Tietomallinnus",
        "tila": "ODOTTAA",
      },
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.2830496143.3575999363",
        "jarjestys": 2,
        "kategoria": "Yleinen",
        "tila": "ODOTTAA",
      },
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.3084966615.4150346995",
        "jarjestys": 3,
        "kategoria": "Yleinen",
        "tila": "ODOTTAA",
      },
    ],
    "esitettavatYhteystiedot": Array [
      Object {
        "__typename": "Yhteystieto",
        "etunimi": "Marko",
        "organisaatio": "Kajaani",
        "puhelinnumero": "0293121213",
        "sahkoposti": "markku.koi@koi.com",
        "sukunimi": "Koi",
      },
    ],
    "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
    "videot": Array [
      Object {
        "__typename": "Linkki",
        "nimi": "Esittely 1",
        "url": "https://video",
      },
    ],
    "vuorovaikutusJulkaisuPaiva": "2022-03-23",
    "vuorovaikutusNumero": 1,
    "vuorovaikutusTilaisuudet": Array [
      Object {
        "__typename": "VuorovaikutusTilaisuus",
        "alkamisAika": "15:00",
        "kaytettavaPalvelu": "TEAMS",
        "linkki": "https://linkki_tilaisuuteen",
        "nimi": "Lorem ipsum 1",
        "paattymisAika": "16:00",
        "paivamaara": "2022-03-04",
        "tyyppi": "VERKOSSA",
      },
      Object {
        "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
        "__typename": "VuorovaikutusTilaisuus",
        "alkamisAika": "10:00",
        "nimi": "Lorem ipsum two 1",
        "osoite": "Katu 123",
        "paattymisAika": "11:00",
        "paikka": "Kunnantalo",
        "paivamaara": "2022-04-05",
        "postinumero": "00100",
        "postitoimipaikka": "Helsinki",
        "tyyppi": "PAIKALLA",
      },
    ],
    "vuorovaikutusYhteysHenkilot": Array [
      "A000112",
    ],
  },
}
`;

exports[`Api should search, load and save a project 5`] = `
Object {
  "description": "saveAndVerifyAineistoSave",
  "obj": Object {
    "__typename": "Vuorovaikutus",
    "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
    "aineistot": Array [
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.2303050596.2160380418",
        "jarjestys": 11,
        "kategoria": "Tietomallinnus new",
        "tila": "ODOTTAA",
      },
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.2830496143.3575999363",
        "jarjestys": 12,
        "kategoria": "Yleinen new",
        "tila": "ODOTTAA",
      },
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.3084966615.4150346995",
        "jarjestys": 13,
        "kategoria": "Yleinen new",
        "tila": "ODOTTAA",
      },
    ],
    "esitettavatYhteystiedot": Array [
      Object {
        "__typename": "Yhteystieto",
        "etunimi": "Marko",
        "organisaatio": "Kajaani",
        "puhelinnumero": "0293121213",
        "sahkoposti": "markku.koi@koi.com",
        "sukunimi": "Koi",
      },
    ],
    "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
    "videot": Array [
      Object {
        "__typename": "Linkki",
        "nimi": "Esittely 1",
        "url": "https://video",
      },
    ],
    "vuorovaikutusJulkaisuPaiva": "2022-03-23",
    "vuorovaikutusNumero": 1,
    "vuorovaikutusTilaisuudet": Array [
      Object {
        "__typename": "VuorovaikutusTilaisuus",
        "alkamisAika": "15:00",
        "kaytettavaPalvelu": "TEAMS",
        "linkki": "https://linkki_tilaisuuteen",
        "nimi": "Lorem ipsum 1",
        "paattymisAika": "16:00",
        "paivamaara": "2022-03-04",
        "tyyppi": "VERKOSSA",
      },
      Object {
        "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
        "__typename": "VuorovaikutusTilaisuus",
        "alkamisAika": "10:00",
        "nimi": "Lorem ipsum two 1",
        "osoite": "Katu 123",
        "paattymisAika": "11:00",
        "paikka": "Kunnantalo",
        "paivamaara": "2022-04-05",
        "postinumero": "00100",
        "postitoimipaikka": "Helsinki",
        "tyyppi": "PAIKALLA",
      },
    ],
    "vuorovaikutusYhteysHenkilot": Array [
      "A000112",
    ],
  },
}
`;

exports[`Api should search, load and save a project 6`] = `
Object {
  "description": "saveAndVerifyAineistoSave",
  "obj": Object {
    "__typename": "Vuorovaikutus",
    "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
    "aineistot": Array [
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.2830496143.3575999363",
        "jarjestys": 12,
        "kategoria": "Yleinen new",
        "tila": "ODOTTAA",
      },
      Object {
        "__typename": "Aineisto",
        "dokumenttiOid": "1.2.246.578.5.100.3084966615.4150346995",
        "jarjestys": 13,
        "kategoria": "Yleinen new",
        "tila": "ODOTTAA",
      },
    ],
    "esitettavatYhteystiedot": Array [
      Object {
        "__typename": "Yhteystieto",
        "etunimi": "Marko",
        "organisaatio": "Kajaani",
        "puhelinnumero": "0293121213",
        "sahkoposti": "markku.koi@koi.com",
        "sukunimi": "Koi",
      },
    ],
    "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
    "videot": Array [
      Object {
        "__typename": "Linkki",
        "nimi": "Esittely 1",
        "url": "https://video",
      },
    ],
    "vuorovaikutusJulkaisuPaiva": "2022-03-23",
    "vuorovaikutusNumero": 1,
    "vuorovaikutusTilaisuudet": Array [
      Object {
        "__typename": "VuorovaikutusTilaisuus",
        "alkamisAika": "15:00",
        "kaytettavaPalvelu": "TEAMS",
        "linkki": "https://linkki_tilaisuuteen",
        "nimi": "Lorem ipsum 1",
        "paattymisAika": "16:00",
        "paivamaara": "2022-03-04",
        "tyyppi": "VERKOSSA",
      },
      Object {
        "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
        "__typename": "VuorovaikutusTilaisuus",
        "alkamisAika": "10:00",
        "nimi": "Lorem ipsum two 1",
        "osoite": "Katu 123",
        "paattymisAika": "11:00",
        "paikka": "Kunnantalo",
        "paivamaara": "2022-04-05",
        "postinumero": "00100",
        "postitoimipaikka": "Helsinki",
        "tyyppi": "PAIKALLA",
      },
    ],
    "vuorovaikutusYhteysHenkilot": Array [
      "A000112",
    ],
  },
}
`;

exports[`Api should search, load and save a project 7`] = `
Array [
  Object {
    "Records": Array [
      Object {
        "body": "{\\"oid\\":\\"1.2.246.578.5.1.2978288874.2711575506\\",\\"vuorovaikutusNumero\\":1}",
      },
    ],
  },
]
`;

exports[`Api should search, load and save a project 8`] = `
Object {
  "__typename": "Vuorovaikutus",
  "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
  "aineistot": Array [
    Object {
      "__typename": "Aineisto",
      "dokumenttiOid": "1.2.246.578.5.100.2830496143.3575999363",
      "jarjestys": 12,
      "kategoria": "Yleinen new",
      "tiedosto": "/suunnitteluvaihe/vuorovaikutus_1/ekatiedosto_eka.pdf",
      "tila": "VALMIS",
      "tuotu": "***unittest***",
    },
    Object {
      "__typename": "Aineisto",
      "dokumenttiOid": "1.2.246.578.5.100.3084966615.4150346995",
      "jarjestys": 13,
      "kategoria": "Yleinen new",
      "tiedosto": "/suunnitteluvaihe/vuorovaikutus_1/karttakuvalla_tiedosto.pdf",
      "tila": "VALMIS",
      "tuotu": "***unittest***",
    },
  ],
  "esitettavatYhteystiedot": Array [
    Object {
      "__typename": "Yhteystieto",
      "etunimi": "Marko",
      "organisaatio": "Kajaani",
      "puhelinnumero": "0293121213",
      "sahkoposti": "markku.koi@koi.com",
      "sukunimi": "Koi",
    },
  ],
  "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
  "videot": Array [
    Object {
      "__typename": "Linkki",
      "nimi": "Esittely 1",
      "url": "https://video",
    },
  ],
  "vuorovaikutusJulkaisuPaiva": "2022-03-23",
  "vuorovaikutusNumero": 1,
  "vuorovaikutusTilaisuudet": Array [
    Object {
      "__typename": "VuorovaikutusTilaisuus",
      "alkamisAika": "15:00",
      "kaytettavaPalvelu": "TEAMS",
      "linkki": "https://linkki_tilaisuuteen",
      "nimi": "Lorem ipsum 1",
      "paattymisAika": "16:00",
      "paivamaara": "2022-03-04",
      "tyyppi": "VERKOSSA",
    },
    Object {
      "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
      "__typename": "VuorovaikutusTilaisuus",
      "alkamisAika": "10:00",
      "nimi": "Lorem ipsum two 1",
      "osoite": "Katu 123",
      "paattymisAika": "11:00",
      "paikka": "Kunnantalo",
      "paivamaara": "2022-04-05",
      "postinumero": "00100",
      "postitoimipaikka": "Helsinki",
      "tyyppi": "PAIKALLA",
    },
  ],
  "vuorovaikutusYhteysHenkilot": Array [
    "A000112",
  ],
}
`;

exports[`Api should search, load and save a project 9`] = `
Object {
  "description": "publicProjekti ennen suunnitteluvaihetta",
  "obj": Object {
    "__typename": "ProjektiJulkinen",
    "aloitusKuulutusJulkaisut": Array [
      Object {
        "__typename": "AloitusKuulutusJulkaisuJulkinen",
        "aloituskuulutusPDFt": Object {
          "SUOMI": Object {
            "aloituskuulutusIlmoitusPDFPath": "/tiedostot/suunnitelma/1.2.246.578.5.1.2978288874.2711575506/aloituskuulutus/ILMOITUS TOIMIVALTAISEN VIRANOMAISEN KUULUTUKSESTA HASSU AUTOMAATTITESTIPROJEKTI1.pdf",
            "aloituskuulutusPDFPath": "/tiedostot/suunnitelma/1.2.246.578.5.1.2978288874.2711575506/aloituskuulutus/KUULUTUS SUUNNITTELUN ALOITTAMISESTA HASSU AUTOMAATTITESTIPROJEKTI1.pdf",
          },
          "__typename": "AloitusKuulutusPDFt",
        },
        "elyKeskus": "Pirkanmaa",
        "hankkeenKuvaus": Object {
          "RUOTSI": "P책 Svenska",
          "SAAME": "Saameksi",
          "SUOMI": "Lorem Ipsum",
          "__typename": "HankkeenKuvaukset",
        },
        "kielitiedot": Object {
          "__typename": "Kielitiedot",
          "ensisijainenKieli": "SUOMI",
        },
        "kuulutusPaiva": "2022-01-02",
        "siirtyySuunnitteluVaiheeseen": "2022-01-01",
        "suunnitteluSopimus": Object {
          "__typename": "SuunnitteluSopimus",
          "email": "Joku.Jossain@vayla.fi",
          "etunimi": "Joku",
          "kunta": "Nokia",
          "logo": "/tiedostot/suunnitelma/1.2.246.578.5.1.2978288874.2711575506/suunnittelusopimus/logo.png",
          "puhelinnumero": "123",
          "sukunimi": "Jossain",
        },
        "velho": Object {
          "__typename": "Velho",
          "kunnat": Array [
            "Helsinki",
            " Vantaa",
          ],
          "nimi": "HASSU AUTOMAATTITESTIPROJEKTI1",
          "tilaajaOrganisaatio": "V채yl채virasto",
          "tyyppi": "TIE",
          "vaylamuoto": Array [
            "tie",
          ],
        },
        "yhteystiedot": Array [
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "A-tunnus1",
            "organisaatio": "CGI Suomi Oy",
            "puhelinnumero": "123",
            "sahkoposti": "mikko.haapamki@cgi.com",
            "sukunimi": "Hassu",
          },
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "Marko",
            "organisaatio": "Kajaani",
            "puhelinnumero": "0293121213",
            "sahkoposti": "markku.koi@koi.com",
            "sukunimi": "Koi",
          },
        ],
      },
    ],
    "euRahoitus": false,
    "oid": "1.2.246.578.5.1.2978288874.2711575506",
    "status": "ALOITUSKUULUTUS",
  },
}
`;

exports[`Api should search, load and save a project 10`] = `
Object {
  "description": "projekti palaute lis채tty",
  "obj": Array [
    Object {
      "__typename": "Palaute",
      "etunimi": "Matti",
      "id": "***unittest***",
      "kysymysTaiPalaute": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "liite": "/palautteet/***unittest***/logo.png",
      "puhelinnumero": "123456",
      "sahkoposti": "test@vayla.fi",
      "sukunimi": "Meikalainen",
      "yhteydenottotapaEmail": false,
      "yhteydenottotapaPuhelin": true,
    },
  ],
}
`;

exports[`Api should search, load and save a project 11`] = `
Object {
  "description": "projekti palaute otettu k채sittelyyn",
  "obj": Array [
    Object {
      "__typename": "Palaute",
      "etunimi": "Matti",
      "id": "***unittest***",
      "kysymysTaiPalaute": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "liite": "/palautteet/***unittest***/logo.png",
      "otettuKasittelyyn": true,
      "puhelinnumero": "123456",
      "sahkoposti": "test@vayla.fi",
      "sukunimi": "Meikalainen",
      "yhteydenottotapaEmail": false,
      "yhteydenottotapaPuhelin": true,
    },
  ],
}
`;

exports[`Api should search, load and save a project 12`] = `
Object {
  "description": "publicProjekti suunnitteluvaihe",
  "obj": Object {
    "__typename": "SuunnitteluVaiheJulkinen",
    "arvioSeuraavanVaiheenAlkamisesta": "huomenna",
    "hankkeenKuvaus": Object {
      "SAAME": "Saameksi suunnitteluvaihe",
      "SUOMI": "Lorem Ipsum suunnitteluvaihe",
      "__typename": "HankkeenKuvaukset",
    },
    "suunnittelunEteneminenJaKesto": "suunnitelma etenee aikataulussa ja valmistuu vuoden 2022 aikana",
    "vuorovaikutukset": Array [
      Object {
        "__typename": "VuorovaikutusJulkinen",
        "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
        "aineistot": Array [
          Object {
            "__typename": "Aineisto",
            "dokumenttiOid": "1.2.246.578.5.100.2830496143.3575999363",
            "jarjestys": 12,
            "kategoria": "Yleinen new",
            "tiedosto": "/suunnitteluvaihe/vuorovaikutus_1/ekatiedosto_eka.pdf",
            "tila": "VALMIS",
            "tuotu": "***unittest***",
          },
          Object {
            "__typename": "Aineisto",
            "dokumenttiOid": "1.2.246.578.5.100.3084966615.4150346995",
            "jarjestys": 13,
            "kategoria": "Yleinen new",
            "tiedosto": "/suunnitteluvaihe/vuorovaikutus_1/karttakuvalla_tiedosto.pdf",
            "tila": "VALMIS",
            "tuotu": "***unittest***",
          },
        ],
        "esitettavatYhteystiedot": Array [
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "Marko",
            "organisaatio": "Kajaani",
            "puhelinnumero": "0293121213",
            "sahkoposti": "markku.koi@koi.com",
            "sukunimi": "Koi",
          },
        ],
        "julkinen": true,
        "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
        "videot": Array [
          Object {
            "__typename": "Linkki",
            "nimi": "Esittely 1",
            "url": "https://video",
          },
        ],
        "vuorovaikutusJulkaisuPaiva": "2022-03-23",
        "vuorovaikutusNumero": 1,
        "vuorovaikutusTilaisuudet": Array [
          Object {
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "15:00",
            "kaytettavaPalvelu": "TEAMS",
            "linkki": "https://linkki_tilaisuuteen",
            "nimi": "Lorem ipsum 1",
            "paattymisAika": "16:00",
            "paivamaara": "2022-03-04",
            "tyyppi": "VERKOSSA",
          },
          Object {
            "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "10:00",
            "nimi": "Lorem ipsum two 1",
            "osoite": "Katu 123",
            "paattymisAika": "11:00",
            "paikka": "Kunnantalo",
            "paivamaara": "2022-04-05",
            "postinumero": "00100",
            "postitoimipaikka": "Helsinki",
            "tyyppi": "PAIKALLA",
          },
        ],
        "vuorovaikutusYhteysHenkilot": Array [
          "A000112",
        ],
        "vuorovaikutusYhteystiedot": Array [
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "A-tunnus1",
            "organisaatio": "CGI Suomi Oy",
            "puhelinnumero": "123",
            "sahkoposti": "mikko.haapamki@cgi.com",
            "sukunimi": "Hassu",
          },
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "Marko",
            "organisaatio": "Kajaani",
            "puhelinnumero": "0293121213",
            "sahkoposti": "markku.koi@koi.com",
            "sukunimi": "Koi",
          },
        ],
      },
      Object {
        "__typename": "VuorovaikutusJulkinen",
        "aineistoPoistetaanNakyvista": "2222-03-23T23:45",
        "aineistot": undefined,
        "esitettavatYhteystiedot": Array [
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "Marko",
            "organisaatio": "Kajaani",
            "puhelinnumero": "0293121213",
            "sahkoposti": "markku.koi@koi.com",
            "sukunimi": "Koi",
          },
        ],
        "kysymyksetJaPalautteetViimeistaan": "2022-03-23T23:48",
        "videot": Array [
          Object {
            "__typename": "Linkki",
            "nimi": "Esittely 2",
            "url": "https://video",
          },
        ],
        "vuorovaikutusJulkaisuPaiva": "2022-03-23",
        "vuorovaikutusNumero": 2,
        "vuorovaikutusTilaisuudet": Array [
          Object {
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "15:00",
            "kaytettavaPalvelu": "TEAMS",
            "linkki": "https://linkki_tilaisuuteen",
            "nimi": "Lorem ipsum 2",
            "paattymisAika": "16:00",
            "paivamaara": "2022-03-04",
            "tyyppi": "VERKOSSA",
          },
          Object {
            "Saapumisohjeet": "Ensimm채inen ovi vasemmalla",
            "__typename": "VuorovaikutusTilaisuus",
            "alkamisAika": "10:00",
            "nimi": "Lorem ipsum two 2",
            "osoite": "Katu 123",
            "paattymisAika": "11:00",
            "paikka": "Kunnantalo",
            "paivamaara": "2022-04-05",
            "postinumero": "00100",
            "postitoimipaikka": "Helsinki",
            "tyyppi": "PAIKALLA",
          },
        ],
        "vuorovaikutusYhteysHenkilot": Array [
          "A000112",
          "FOO123",
        ],
        "vuorovaikutusYhteystiedot": Array [
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "A-tunnus1",
            "organisaatio": "CGI Suomi Oy",
            "puhelinnumero": "123",
            "sahkoposti": "mikko.haapamki@cgi.com",
            "sukunimi": "Hassu",
          },
          Object {
            "__typename": "Yhteystieto",
            "etunimi": "Marko",
            "organisaatio": "Kajaani",
            "puhelinnumero": "0293121213",
            "sahkoposti": "markku.koi@koi.com",
            "sukunimi": "Koi",
          },
        ],
      },
    ],
  },
}
`;
